### ISSUE TITLE
Check available disk space before build

### ISSUE THREAD

**Issue by matthiasrichter** [alisw/alibuild#434]
ROOT 6 compilation with the default CXXFLAGS needs >13 GB disk space for compilation. That's an extreme case, but in any way checking the disk space before starting the build would be a nice feature. It can be an interactive warning and the user can decide whether to continue or not.

How about adding an optional required size estimate to recipes? I know, the estimation might be tricky, still we can gain a lot of guidance for the end user.

**Comment by ktf** [comment]
Maybe we should actually try to trim down ROOT6 installation, detecting correct available space can be tricky (e.g. on network filesystems). Can you do a `du -sh` of the root installation?

**Comment by dberzano** [comment]
I agree with @ktf. @pzhristov has found some ways to trim down the ROOT 6 compilation (which is very large due to debug symbols in the internal LLVM), maybe he can comment on that.

**Comment by pzhristov** [comment]
I am attaching the small patch that removes the "-g" from the compilation of the interpreter, if we are in "Release" mode.
[patch.txt](https://github.com/alisw/alibuild/files/1231098/patch.txt)

**Comment by matthiasrichter** [comment]
but that looks like an inconsistency in the CMake setup. The '-g' should not be there in the first place if the build type is 'Release'. Maybe it's better to define proper compilation flags according to build type.

**Comment by pzhristov** [comment]
The problem is that interpreter uses the global CMAKE_CXX_FLAGS and ignores the "Release" build type for clang. So Root is compiled in "RELWITHDEBINFO" and we want the interpreter in "Release".

**Comment by dberzano** [comment]
@matthiasrichter as @pzhristov said, this is indeed an inconsistency - and should be fixed in ROOT itself. Can we close this bug as it has nothing to do with aliBuild? :-)

Let's open a new issue in alidist saying to upgrade to the new ROOT as soon as they release it with the patch that Peter has attached.

**Comment by sawenzel** [comment]
@dberzano : The inconsistency does not come from ROOT I believe, it is because we are enforcing "-g" in our defaults.

**Comment by pzhristov** [comment]
@sawenzel, it is coming from Root since interpreter/CMakeLists  ignores the LLVM_BUILD_TYPE when it uses directly the CMAKE_CXX_FLAGS. What we want is RELWITHDEBINFO everywhere in Root, except in interpreter where we need Release.

**Comment by sawenzel** [comment]
@pzhristov , @dberzano : Without alibuild, I am running for ROOT (master)
`cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DLLVM_BUILD_TYPE=Release -DCMAKE_EXPORT_COMPILE_COMMANDS=ON`

Inspecting the compilations database tells me that everything is correct and I don't see -g for LLVM.

If however, you do (as alibuild is doing):

`CXXFLAGS=-g cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DLLVM_BUILD_TYPE=Release -DCMAKE_EXPORT_COMPILE_COMMANDS=ON`

we get the problem because the external CXXFLAGS is influencing everything. So the origin of the problem comes from globally setting -g in CXXFLAGS. I think it is this choice that we need to rethink.

**Comment by sawenzel** [comment]
See also existing issue on this topic https://github.com/alisw/alibuild/issues/400 for further discussion.

**Comment by matthiasrichter** [comment]
I have checked the case, it's still valid. @dberzano the issue should be reopened.

My user story: As a developer I want to build all packages with build type RELEASE, except some packages I'm working on. For those I want to write a custom build type in the defaults file, like e.g.
```
diff --git a/defaults-o2.sh b/defaults-o2.sh
index 2027e5e..1cf9042 100644
--- a/defaults-o2.sh
+++ b/defaults-o2.sh
@@ -43,6 +43,9 @@ overrides:
       - GEANT4_VMC
       - Vc
       - ZeroMQ
+  FairRoot:
+    env:
+      CMAKE_BUILD_TYPE: "RELWITHDEBINFO"
   GSL:
```
expecting now that the environment from the overrides section is propagated to the build of my package. This does not happen.

Note: For the recipes I'm using a patch which replace the hardcoded build type in the recipe
https://github.com/matthiasrichter/alidist/commit/ed0a8daa498777cf0c9e5c351871673d0a916f20

**Comment by matthiasrichter** [comment]
@dberzano sorry, ended up in the wrong issue, I will re post in the correct one.
